sys-kernel/gentoo-kernel/gentoo-kernel-6.3.6-r1.ebuild

```diff
@@ -136,5 +136,81 @@ src_prepare() {
 		merge_configs+=( "${dist_conf_path}/big-endian.config" )
 	fi
 
+	if use modules-sign; then
+		: "${MODULES_SIGN_HASH:=sha512}"
+		cat <<-EOF > "${dist_conf_path}/modules-sign.config" || die
+			## Enable module signing
+			CONFIG_MODULE_SIG=y
+			CONFIG_MODULE_SIG_ALL=y
+			CONFIG_MODULE_SIG_FORCE=y
+			CONFIG_MODULE_SIG_${MODULES_SIGN_HASH^^}=y
+		EOF
+		if [ -n "${MODULES_SIGN_KEY}" ]; then
+			if [ -f "${MODULES_SIGN_KEY}" ]; then
+				echo "CONFIG_MODULE_SIG_KEY=\"${MODULES_SIGN_KEY}\"" \
+					>> "${dist_conf_path}/modules-sign.config"
+			else
+				die "MODULES_SIGN_KEY=${MODULES_SIGN_KEY} not found!"
+			fi
+		fi
+		merge_configs+=( "${dist_conf_path}/modules-sign.config" )
+	fi
+
 	kernel-build_merge_configs "${merge_configs[@]}"
 }
+
+src_install() {
+	kernel-build_src_install
+	if use modules-sign; then
+		local dir_ver=${PV}${KV_LOCALVERSION}
+		local kernel_dir=/usr/src/linux-${dir_ver}
+
+		# Copy the key/certificate so external modules can be signed
+		# If MODULES_SIGN_KEY is set no key was generated
+		if [ -z "${MODULES_SIGN_KEY}" ]; then
```

Comment:
Also, when unset, in linux-mod-r1 MODULES_SIGN_KEY defaults to whatever is set in the kernel (users could've set through config.d or USE=savedconfig), but it's ignored here.

Reply:
Why would the paths be wrong? When I tried setting `CONFIG_MODULE_SIG_KEY=` earlier it seemed that no key is generated at all if it is set to anything but the default value. i.e. if it is set the build system appears to expect that the key is already there and does not generate one. So if a key was generated then this must be the path.

Reply:
If this variable is unset but the kernel setting was changed, then it'll try to copy a path that does not exist below.

Reply:
(not that this matters anymore given that code is gone)